src/components/PlatformNavigator/index.tsx

```diff
@@ -52,11 +64,34 @@ export function PlatformNavigator({
               {frameworks.map((platform, index) => {
                 const title = platform.title;
                 const current = title === platformTitle;
+                let href: LinkProps['href'];
+
+                /**
+                 * If this platform in the list exists for the current pathname,
+                 * we'll link to that platforms version of the page.
+                 */
+
+                if (
+                  allowedPlatforms.includes(platform.key) &&
+                  pathname !== '/gen1'
```

Comment:
The /gen1/index.tsx page is a little weird because we list platforms on it which get built into the directory.json, but the page itself isn't part of the [platform] query folder. I tried removing those platforms, but it breaks the NextPrevious component. Without this `pathname !== '/gen1'` check, it attempts to put a ?platform=react query param on those pages which creates a link that does nothing since /gen1?platform=swift just goes to /gen1 which shows React content.